xformers/helpers/hierarchical_configs.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+from xformers.components.residual import LayerNormStyle
+
+
+@dataclass
+class BasicLayerConfig:
+    embedding: int
+    n_heads: int
+    pooling_ratio: int
+    attention_mechanism: str
+
+
+def get_hierarchical_configuration(
```

Comment:
WIP actually, should not be there probably, see next PR :) (will generate all Metaformers)